src/lib.rs

```diff
@@ -404,16 +400,341 @@
+//! # }
+//! # use rand_core::{OsRng, RngCore};
+//! # let mut client_rng = OsRng;
+//! # let client_registration_start_result = ClientRegistration::<Default>::start(
+//! #     &mut client_rng,
+//! #     b"password",
+//! # )?;
+//! # let mut server_rng = OsRng;
+//! // During setup, server generates its static keypair
+//! let server_kp = Default::generate_random_keypair(&mut server_rng)?;
+//! # let server_registration_start_result = ServerRegistration::<Default>::start(&mut server_rng, client_registration_start_result.message, server_kp.public())?;
+//!
+//! // During setup or registration, the server transmits its static public key to the client
+//! let server_s_pk = server_kp.public(); // obtained from the server
+//! # let client_registration_finish_result = client_registration_start_result.state.finish(&mut client_rng, server_registration_start_result.message, ClientRegistrationFinishParameters::default())?;
+//! # let password_file_bytes = server_registration_start_result.state.finish(client_registration_finish_result.message)?.to_bytes();
+//! # let client_login_start_result = ClientLogin::<Default>::start(
+//! #     &mut client_rng,
+//! #     b"password",
+//! #     ClientLoginStartParameters::default(),
+//! # )?;
+//! # use std::convert::TryFrom;
+//! # let password_file =
+//! #   ServerRegistration::<Default>::try_from(
+//! #     &password_file_bytes[..],
+//! #   )?;
+//! # let server_login_start_result =
+//! #     ServerLogin::start(&mut server_rng, password_file, &server_kp.private(), client_login_start_result.message, ServerLoginStartParameters::default())?;
+//!
+//! // And then later, during login...
+//! let client_login_finish_result = client_login_start_result.state.finish(
+//!     server_login_start_result.message,
+//!     ClientLoginFinishParameters::default(),
+//! )?;
+//!
+//! // Check that the server's static public key matches what was obtained during
+//! // setup or registration
+//! assert_eq!(
+//!     &client_login_finish_result.server_s_pk,
+//!     server_s_pk,
+//! );
+//! # Ok::<(), ProtocolError>(())
+//! ```
+//!
+//! Note that without this check over the consistency of the server's static public key, a malicious actor could impersonate the registration server if it were able to copy the password
+//! file output during registration! Therefore, it is recommended to perform the following check in the application layer if the client can obtain a copy of the server's static
+//! public key beforehand.
+//!
+//!
+//! ## Export Key
+//!
+//! The export key is a pseudorandomly distributed 32-byte string output by both the
+//! [Client Registration Finish](#client-registration-finish) and [Client Login Finish](#client-login-finish) steps.
+//! The same export key string will be output by both functions only if the exact same password is passed to [ClientRegistration::start] and [ClientLogin::start].
+//!
+//! The export key retains as much secrecy as the password itself. Hence, only the parties which
```

Comment:
nit: An allusion to (optional) password hashing being in the pipeline between key and password would give more context as to the feasibility of brute-force attacks. Even better if we're able to refer to another section of our doc for details on export key derivation.